pkg/volume/gcepd/attacher.go

```diff
@@ -313,12 +318,15 @@ func (attacher *gcePersistentDiskAttacher) MountDevice(spec *volume.Spec, device
 		return err
 	}
 
-	options := []string{}
+	// because we want to speed up mkfs, we're not going to discard used blocks at mkfs time, which is fine
+	// for PD if the disk is unused before and discard option is passed in at mount time
+	options := []string{discardOnMount}
 	if readOnly {
 		options = append(options, "ro")
 	}
 	if notMnt {
 		diskMounter := volumeutil.NewSafeFormatAndMountFromHost(gcePersistentDiskPluginName, attacher.host)
+		diskMounter.MkfsNodiscard = true
 		mountOptions := volumeutil.MountOptionFromSpec(spec, options...)
```

Comment:
User can also set custom mount options through the PV. We may need to handle a user specifying "nodiscard".

Reply:
I didn't see where the users' options are read by the mounter? would you mind giving me a pointer to it?

Reply:
I don't think any of user specified mount options are applied during `mkfs`. So are we saying if user specifies `nodiscard` we must override it?

Reply:
The `MountOptionsFromSpec` call right here grabs mount options user can set on PV

Reply:
We were sort of discussing on slack. If `discard` should be a default mount option for gce-pd, isn't it better to document it somewhere and make sure CO versions(such as GKE, Openshift) ship with storageclasses where this option is set? If we move the option in code then - it will require code change to fix it (if it breaks something unrelated).